### ISSUE TITLE
wicketd/MGS - Get real inventory from MGS and serve it from wicketd

### ISSUE THREAD

**Issue by andrewjstone** [oxidecomputer/omicron#1870]
This is the first step in runtime integration with MGS. We want to get real inventory data from MGS. We also want to support dummy versions for when we aren't running on a real rack, so wicket can be tested in a self-contained manner.

**Comment by plotnick** [comment]
If we can do so without slowing down development, I'd like to coordinate this work (and possibly other pieces, like #1867, etc.) with the nascent [RFD 334](https://rfd.shared.oxide.computer/rfd/0334).

**Comment by andrewjstone** [comment]
Let's do it @plotnick! It'll be great to have a partner in crime on this. With @jgallagher I suppose it would be a Triumverate :)

**Comment by andrewjstone** [comment]
Inventory will now be retrieved from MGS by wicketd. I'm changing the title to reflect this.

**Comment by andrewjstone** [comment]
The first PR is out for review: https://github.com/oxidecomputer/omicron/pull/1912

**Comment by andrewjstone** [comment]
Another PR out for review: https://github.com/oxidecomputer/omicron/pull/1930

**Comment by candidobrau** [comment]
PRs 1912 and 1930 have been reviewed and merged. There is some work going on with `progenitor` that will be helpful. @andrewjstone will follow up on when that progenitor work will be done.

**Comment by andrewjstone** [comment]
The progenitor work is done, and used in https://github.com/oxidecomputer/omicron/pull/1953

**Comment by andrewjstone** [comment]
Once https://github.com/oxidecomputer/omicron/pull/1953 is merged in we can close this. There are a few other issues I will open individually.

**Comment by andrewjstone** [comment]
Done enough to consider this closed. Small related issues are opened elsewhere.
